diff/walking/differ.go

```diff
@@ -1,154 +1,44 @@
 package walking
 
 import (
+	"context"
 	"crypto/rand"
 	"encoding/base64"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"os"
-	"strings"
 	"time"
 
 	"github.com/containerd/containerd/archive"
 	"github.com/containerd/containerd/archive/compression"
 	"github.com/containerd/containerd/content"
 	"github.com/containerd/containerd/diff"
 	"github.com/containerd/containerd/errdefs"
-	"github.com/containerd/containerd/images"
 	"github.com/containerd/containerd/log"
-	"github.com/containerd/containerd/metadata"
 	"github.com/containerd/containerd/mount"
-	"github.com/containerd/containerd/platforms"
-	"github.com/containerd/containerd/plugin"
 	digest "github.com/opencontainers/go-digest"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/pkg/errors"
-	"github.com/sirupsen/logrus"
-	"golang.org/x/net/context"
 )
 
-func init() {
-	plugin.Register(&plugin.Registration{
-		Type: plugin.DiffPlugin,
-		ID:   "walking",
-		Requires: []plugin.Type{
-			plugin.MetadataPlugin,
-		},
-		InitFn: func(ic *plugin.InitContext) (interface{}, error) {
-			md, err := ic.Get(plugin.MetadataPlugin)
-			if err != nil {
-				return nil, err
-			}
-
-			ic.Meta.Platforms = append(ic.Meta.Platforms, platforms.DefaultSpec())
-			return NewWalkingDiff(md.(*metadata.DB).ContentStore())
-		},
-	})
-}
-
 type walkingDiff struct {
 	store content.Store
 }
 
 var emptyDesc = ocispec.Descriptor{}
 
-// NewWalkingDiff is a generic implementation of diff.Differ.
+// NewWalkingDiff is a generic implementation of diff.Comparer.
 // NewWalkingDiff is expected to work with any filesystem.
-func NewWalkingDiff(store content.Store) (diff.Differ, error) {
+func NewWalkingDiff(store content.Store) diff.Comparer {
```

Comment:
With the new interface names, should this be `NewWalkingCompare()` ?

unrelated nit: the docstring could provide more information about what makes this implementation different from other potential implementations.

Reply:
Updated the doc string, kept name to better align with package `diff/walking` rather than interface. Yes, the names aren't great...